src/main/java/redis/clients/jedis/DefaultJedisSocketFactory.java

```diff
@@ -121,4 +121,12 @@ public int getSoTimeout() {
   public void setSoTimeout(int soTimeout) {
     this.soTimeout = soTimeout;
   }
+
+  @Override
+  public JedisSocketFactory copy() {
```

Comment:
Why copy and not clone and clonable?

Reply:
I have always found clone/Cloneable not that helpful, specially in its contract:
* Where it strongly insist that you shouldn't call any constructors
* It throws `CloneNotSupportedException` and I don't want that propagating everywhere (although that can be suppress in subclass with try/catch/throw runtime but that's usually code smell)
* Objects should implement equality (not really wrong but not all objects need that)

That being said, I am not specifically against clone here if that's something more consistent with the rest of the codebase or if we think it's better for the use case here, what do you think?

Reply:
I think it's better to implement Clonable and clone(), notice you don't have to have to declare CloneNotSupportedException.

And just declare:
```
@override
public JedisSocketFactory clone() {
 return new DefaultJedisSocketFactory(
      this.host, this.port, this.connectionTimeout, this.soTimeout,
      this.ssl, this.sslSocketFactory, this.sslParameters, this.hostnameVerifier
    );
}
```

Reply:
Well, that's the problem I meant, this implementation breaks the `clone` contract:
1. It calls a constructor
2. It doesn't call `super.clone()`

The implementation of clone that doesn't break the `Cloneable` contract would be:

```java
@Override
public DefaultJedisSocketFactory clone() throws CloneNotSupportedException {
  // forced to throw above because we need to call super.clone()
  DefaultJedisSocketFactory clone = (DefaultJedisSocketFactory) super.clone();
  clone.soTimeout = 123; // Apply any additional settings through fields directly (this line should go away, it's just here as an example)
  return clone;
}
```
```

Reply:
@gkorland pinging here as a reminder, what do you think about the above?